### PR TITLE
chore: refactor server code

### PR DESCRIPTION
**Pull request by swarit-pandey**
Purpose: The current code can be potentially improved to use standard REST based structure used in Go projects. Currently there is no scope of dependency injection, seperation of concerned layers which could potentially create issues going ahead (as we add more routes and businees logic). 

This PR overhauls a lot of code that was implemented previously (excpet core logic). The new changes introduces:
- Seperation of concerns with proper constructors
- Not letting server crash when errors are encountered instead return them to clients 
- Interface based layer initialization for better integration testing (as mocking interfaces is super easy, while structs are concrete)

Potential improvements:
- Routes can be moved to a seperate package 
- A more dedicated library can be used such as Gin 
- Errors should be properly wrapped before reaching clients

Caveats:
This is not to undermine the work previous contributors did, if the reviewers/maintainers think that this change is 'very drastic', feel free to close the PR. But since a lot of service layer code is still to be implemented we can potentially improve the code. 

Thanks.

### CODE REVIEW

**Comment by lucifercr07** [comment]
@swarit-pandey thanks for this effort, please resolve the conflicts nce.

**Comment by swarit-pandey** [comment]
>@swarit-pandey thanks for this contribution, please resolve the conflicts once.

@lucifercr07 Sure, will get it done by EOD today.

File: config/config.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ type Config struct {
 }
 
 // LoadConfig loads the application configuration from environment variables or defaults
```
Comment (santhosh-chinnasamy) [inline config/config.go:16]:
```suggestion
// loads the application configuration from environment variables or defaults
```

Comment (swarit-pandey) [reply config/config.go]: In Go comments if a function is public then it should have a comment starting with its name.

File: internal/db/dicedb.go [hunk 2]
```diff
@@ -1,102 +0,0 @@
...
-	return &DiceDB{
-		Client: diceClient,
-		Ctx:    context.Background(),
-	}, nil
-}
-
-// ExecuteCommand executes a command based on the input
-func (db *DiceDB) ExecuteCommand(command *cmds.CommandRequest) (interface{}, error) {
-	switch command.Cmd {
-	case "GET":
```
Comment (lucifercr07) [inline internal/db/dicedb.go:60]: Please match the behaviour of current changes similar to this.

File: internal/cmds/cmds.go [hunk 3]
```diff
@@ -4,3 +4,9 @@ type CommandRequest struct {
 	Cmd  string `json:"cmd"`
 	Args []string
 }
+
+type CommandRequestArgs struct {
```
Comment (lucifercr07) [inline internal/cmds/cmds.go:8]: Not required

File: internal/service/health.go [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package service
+
+// TODO: Implement me
```
Comment (lucifercr07) [inline internal/service/health.go:3]: Why are these empty?

File: internal/repository/request.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package repository
+
+type GetRequest struct {
```
Comment (lucifercr07) [inline internal/repository/request.go:3]: We need to keep this generic, not feasible to be done for multiple commands getting generated.

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

**Comment by lucifercr07** [comment]
Closing the PR, since inactive from sometime.

**Comment by swarit-pandey** [comment]
Hey folks, sorry for not being able to continue working on this. I got occupied with other work related stuff, thanks for taking time to review.

**Pull request closed without merging**